src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -140,6 +154,15 @@ private static function compilePattern(Route $route, $pattern, $isHost)
                     // directly adjacent, e.g. '/{x}{y}'.
                     $regexp .= '+';
                 }
+            } else {
+                if (!preg_match('//u', $regexp)) {
+                    $useUnicode = false;
+                } elseif (!$needsUnicode && preg_match('/[\x80-\xFF]|(?<!\\\\)\\\\(?:\\\\\\\\)*+(?-i:X|[pP][\{CLMNPSZ]|x\{)/', $regexp)) {
```

Comment:
> \xB3 and \x{B3} are equivalent

they are, but as you know, `/\xB3/u` is not equivalent to `/\xB3/` (multibyte utf-8 vs single byte high-ASCII). This is an edge case. In order to not falsly turn on unicode, this makes me think we should auto-enable only when using 3 or more hexadecimal numbers between brackets.